nucypher/blockchain/eth/registry.py

```diff
@@ -88,6 +88,7 @@ def __repr__(self) -> str:
     def id(self) -> str:
         """Returns a hexstr of the registry contents."""
         blake = hashlib.blake2b()
+        blake.update(self.__class__.__name__.encode())
```

Comment:
Oh wow... Was it supposed to be in all the time?

Reply:
I think in production it's impossible to have two different registries classes with the same content, but in test it was all the time. 
Also interesting moment - locally I still have old behavior with ordering which "good" for tests, but one commit changed it for CircleCI

Reply:
Instead of the class name we should use the registry type:
```suggestion
        blake.update(self.REGISTRY_TYPE.encode())
```
We don't want the ID to be different depending if it's a `LocalContractRegistry`, a `TemporaryContractRegistry` or an `InMemoryContractRegistry`, even though all of them are contract registries.

Reply:
while tests we create two same registries Local and Memory and they have same ID, it leads to bugs depending on ordering in cache dictionary. I see this ID not for content but for instance

Reply:
Well, but if this is a problem for tests, then we should monkey-patch it in tests then. I don't see impossible that in production we could have two registries with the same content but different classes (e.g., an InMemory registry persisted to a LocalRegistry)
@KPrasch what do you think?

Reply:
See #1545